### DISCUSSION TITLE
Feedback request: core plugin and open source plugin changes

### DISCUSSION THREAD

**Discussion by GrtFS** [tinymce/tinymce#7420] [General]
Regarding Joe Robinson's BluePrint article ( https://www.tiny.cloud/blog/rich-text-editor-open-source-5-6-0-release/ )

Most people prefer things that are free, but also appreciate that people have to earn a living.  That said...

I've recently added Tiny to a non-profit Filemaker project.  I added the TOC in the last month or so and have realised how amazingly useful it is. Consequently, it's now a part I wouldn't want to surrender.  So if it's removed from Tiny's core plugins, I'll have to look elsewhere, which will be a real shame.

I spent an absurd amount of time trying to find an RTE that fits well with Filemaker and as someone with zero html/css/js skills previously, it was difficult.  Tiny's deep documentation is hugely useful (but why only snippets??!)  as is the Skin tool, which is a tremendous asset and on top of that, just the sheer depth of possibilities Tiny offers.  I have a little trouble with the rigidity of imposed formatting, but understand why it's desired across a wider field.

**So I would put in a little vote, for keeping a TOC 'light' vs available to non-profits/core users.**

(As an aside ... have you looked at the Filemaker market and the business potential for a robust RTE Add-on?  There's a big hole there and Claris opened the door a year ago, when they introduced easy 2-way communication between their built-in 'web viewer' and Filemaker via javascript.  Filemaker is an expensive product and so is mostly used in business.)

**Comment by lostkeys** [comment]
Hey, thanks for your feedback. I'm happy you found good use for the table of contents plugin but sad you found it just at the end of it's life as an open source plugin. It's always a difficult decision taking something away, but every line of code, free or not, has a cost of maintenance to it in some shape or form. We fully understand we will disappoint some people with these changes, but we would not do it if we didn't believe the alternatives were better (which we hope to reveal more details for soon).

In your case, a non-profit program for premium features would likely be a better path forward here, however that is not my table but I'll make my case to those in charge.

Finally, you will still be able to use TinyMCE 5 for quite some time even after TinyMCE 6 has been released.

Thanks,

Fredrik
Product Manager

**Comment by GrtFS** [comment]
Hi Fredrik,

Thank you for the reply.  It is much as I expected and perfectly reasonable.  I shall continue with vs 5... although I'm someone who tends to follow the 'up to date' model and not the 'don't touch anything' model.

"In your case, a non-profit program for premium features would likely be a better path forward here"

What does this involve?  When I say, 'non-profit' it's because it's just for me personally, in a non-business context.  It isn't for a charity organisation, or education etc.

( lostkeys ... my wife's are always in her handbag.  It's pretty much a daily occurrence)
